src/webots/gui/WbSimulationView.cpp

```diff
@@ -746,24 +750,27 @@ void WbSimulationView::realTime() {
   WbSimulationState::instance()->setMode(WbSimulationState::REALTIME);
 }
 
-void WbSimulationView::run() {
-  WbSimulationState::instance()->setMode(WbSimulationState::RUN);
-}
-
 void WbSimulationView::fast() {
   WbSimulationState::instance()->setMode(WbSimulationState::FAST);
 }
 
-void WbSimulationView::disableFastMode(bool disabled) {
-  WbActionManager::instance()->action(WbAction::FAST)->setEnabled(!disabled);
-  mView3D->setUserInteractionDisabled(WbAction::DISABLE_FAST_MODE, disabled);
+void WbSimulationView::disable3DView(bool disabled) {
+  WbActionManager::instance()->action(WbAction::TOGGLE_3D_VIEW)->setEnabled(!disabled);
+  mView3D->setUserInteractionDisabled(WbAction::DISABLE_3D_VIEW, disabled);
 }
 
-void WbSimulationView::updateFastModeOverlay() {
-  if (WbSimulationState::instance()->mode() == WbSimulationState::FAST)
-    renderABlackScreen();
+void WbSimulationView::toggle3DView() {
+  if (WbSimulationState::instance()->is3DViewShown())
+    WbSimulationState::instance()->show3DView(false);
   else
+    WbSimulationState::instance()->show3DView(true);
+}
+
+void WbSimulationView::updateFastModeOverlay() {
```

Comment:
This function should probably be renames as the overlay is no longer a `Fast mode` overlay but a hiddeen 3D view overlay.